controllers/ocsinitialization/ocsinitialization_controller.go

```diff
@@ -332,27 +333,31 @@ func (r *OCSInitializationReconciler) SetupWithManager(mgr ctrl.Manager) error {
 					Name:      util.OcsOperatorConfigName,
 					Namespace: r.OperatorNamespace,
 				},
-			},
-			handler.EnqueueRequestsFromMapFunc(
-				func(context context.Context, obj client.Object) []reconcile.Request {
-					return []reconcile.Request{{
-						NamespacedName: InitNamespacedName(),
-					}}
-				},
-			),
+			}, enqueueOCSInit,
 		).
 		// Watcher for prometheus operator csv
 		Watches(
 			&opv1a1.ClusterServiceVersion{},
-			handler.EnqueueRequestsFromMapFunc(
-				func(context context.Context, obj client.Object) []reconcile.Request {
-					return []reconcile.Request{{
-						NamespacedName: InitNamespacedName(),
-					}}
-				},
-			),
+			enqueueOCSInit,
 			builder.WithPredicates(prometheusPredicate),
+		).
+		Watches(
+			&extv1.CustomResourceDefinition{},
+			enqueueOCSInit,
+			builder.WithPredicates(
+				util.NamePredicate(ClusterClaimCrdName),
+				util.CrdCreateAndDeletePredicate(&r.Log, r.AvailableCrds, ClusterClaimCrdName)),
+			builder.OnlyMetadata)
+
+	if r.AvailableCrds[ClusterClaimCrdName] {
+		ocsInitializationController = ocsInitializationController.Watches(
+			&v1alpha1.ClusterClaim{},
+			enqueueOCSInit,
+			builder.WithPredicates(
+				predicate.GenerationChangedPredicate{},
+				util.NamePredicate(util.OdfInfoNamespacedNameClaimName)),
```

Comment:
Consistency and paring with parenthesis
```suggestion
				util.NamePredicate(util.OdfInfoNamespacedNameClaimName),
			),
```